### DISCUSSION TITLE
On-demand revalidate with middleware rewriting

### DISCUSSION THREAD

**Discussion by maratmuslimov** [vercel/next.js#35968] [Help] [answered]
Hi dears,
I can't figure out what I'm supposed to call inside res.unstable_revalidate() when I use in _middleware subdomain rewrite something like:

url.pathname = `/_pages/${currentHost}${pathname}`;
return NextResponse.rewrite(url);

This page renders by calling sub.domain.com

**Comment by MarkwinVI** [comment]
Struggling with this as well, did you figure out the solution?

**Comment by ijjk** [comment, marked as answer]
Hi, middleware is not run for On-Demand ISR requests so you need to call `res.revalidate()` with the actual page path e.g. `pages/blog/[slug].js` you should call `res.revalidate('/blog/post-1')`.

**Reply by MarkwinVI** [reply]
Hi, thanks for the quick response!

So if Im using middleware to support multi-tenancy for custom domains, all my requests get redirected to this pathname

`    url.pathname = `/_sites/${currentHost} ${pathname}`
`
So that means if I wish to revalidate a page in here I should be calling 
`res.revalidate("/_sites/customdomain")
`
or
`res.revalidate("customdomain")
`
Thanks!

**Reply by ijjk** [reply]
If you have a rewrite in middleware adding the current host you will need to do the same when calling `res.revalidate()` e.g. `res.revalidate('/_sites/curretHost/${pathname}')`. An example of this can be seen in the platforms starter project: https://github.com/vercel/platforms/blob/14a8098ea765560858e81f9629e19780e3ae013c/lib/revalidate.ts#L8

**Reply by MarkwinVI** [reply]
Thanks for your help!

I was passing the right path "/_sites/{customdomain}/path" 

but I was calling the revalidate API from my root domain and not the subdomain for the page I wanted to re-validate.
 
 Once I started calling the API from "https://{customdomain}.com/_sites/{customdomain}" as shown in the example that you gave - it worked!
